ggml-sycl.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "ggml.h"
+#include "ggml-backend.h"
+
+#ifdef  __cplusplus
+extern "C" {
+#endif
+
+#define GGML_SYCL_MAX_DEVICES       16
+#define GGML_SYCL_NAME "SYCL"
+
+GGML_API void   ggml_init_sycl(void);
+GGML_API bool   ggml_sycl_compute_forward(struct ggml_compute_params * params, struct ggml_tensor * tensor);
+GGML_API ggml_backend_t ggml_backend_sycl_init(int device);
+GGML_API ggml_backend_buffer_type_t ggml_backend_sycl_buffer_type(int device);
+GGML_API ggml_backend_buffer_type_t ggml_backend_sycl_host_buffer_type(void);
+GGML_API void   print_sycl_devices();
```

Comment:
- Do not add a function to print the devices, add a set of functions to enumerate them instead
- Use the `ggml_backend_sycl_` prefix for all the backend functions

Device enumeration functions are necessary to support multi GPU with pipeline parallelism. `llm_load_tensors` in llama.cpp also needs to be updated to distribute the layers to the different GPUs.

Reply:
print_sycl_devices() help user to select right device ID.
currently, SYCL don't support multiple GPUs. But it will be handled soon after merge this PR. `llm_load_tensors` will be updated as this suggestion.

Yes, rename it with prefix: `ggml_backend_sycl_`